src/main/scala/beam/sim/BeamHelper.scala

```diff
@@ -780,6 +783,118 @@ trait BeamHelper extends LazyLogging {
+                    PersonUtils.setSex(newPerson, PersonUtils.getSex(person))
+                    PersonUtils.setAge(newPerson, PersonUtils.getAge(person))
+                    PersonUtils.setCarAvail(newPerson, PersonUtils.getCarAvail(person))
+                    PersonUtils.setLicence(newPerson, PersonUtils.getLicense(person))
+                    newPerson.setSelectedPlan(person.getSelectedPlan)
+                    // add the new person to the scenario
+                    scenario.getPopulation.addPerson(newPerson)
+                    Some(newPerson.getId)
+                  } else {
+                    None
+                  }
+                } else {
+                  None
+                }
+              })
+              .toList
+            val vehicles = hh._2.getVehicleIds.asScala.flatMap(v => {
+              val vehicle: Vehicle = scenario.getVehicles.getVehicles.get(v)
+              if (vehicle != null) {
+                // clone the current vehicle to form a new vehicle with different id
+                val newVehicle = scenario.getVehicles.getFactory
+                  .createVehicle(Id.createVehicleId(s"${vehicle.getId.toString}_$i"), vehicle.getType)
+                // add the new cloned vehicle to the scenario
+                scenario.getVehicles.addVehicle(newVehicle)
+                Some(newVehicle.getId)
+              } else {
+                None
+              }
+            })
+            // generate a new household and add the above clone members and vehicles
+            val newHouseHold = scenario.getHouseholds.getFactory
+              .createHousehold(Id.create(s"${hh._1.toString}_$i", classOf[Household]))
+              .asInstanceOf[HouseholdImpl]
+            newHouseHold.setIncome(hh._2.getIncome)
+            if (members.nonEmpty) {
+              newHouseHold.setMemberIds(JavaConverters.seqAsJavaList(members))
+            }
+            if (vehicles.nonEmpty) {
+              newHouseHold.setVehicleIds(JavaConverters.seqAsJavaList(vehicles))
+            }
+            Some(newHouseHold -> hh._2)
+          } else {
+            None
+          }
+        })
+        // add the generated new households with attributes to the current scenario
+        newHouseHolds foreach {
+          case (newhh, oldhh) =>
+            scenario.getHouseholds.getHouseholds.put(newhh.getId, newhh)
+            for (attr <- List("homecoordx", "homecoordy", "housingtype")) {
+              val attrValue = scenario.getHouseholds.getHouseholdAttributes.getAttribute(oldhh.getId.toString, attr)
+              scenario.getHouseholds.getHouseholdAttributes.putAttribute(newhh.getId.toString, attr, attrValue)
+            }
+        }
+      }
+      // reset the counter
+      populationCounter = 0
+      logger.info(s"""After sampling:
+                     |Number of households: ${scenario.getHouseholds.getHouseholds.keySet.size}
+                     |Number of vehicles: ${getVehicleGroupingStringUsing(
```

Comment:
let extract sampling into separate class to save 200+ lines here

Reply:
ok sure

Reply:
Made the changes @wrashid